src/Indigo/Contracts/HomebaseLite/Impl/Contract.hs

```diff
@@ -41,4 +38,9 @@ indigoContract param = defContract $ docGroup "Homebase Lite" do
     )
 
 lorentzContract :: L.Contract Parameter Storage ()
-lorentzContract = defaultContract $ optimize $ unContractCode $ compileIndigoContract indigoContract
+lorentzContract = mkContractWith compilationOptions
+  $ ContractCode . optimize . unContractCode
+  $ compileIndigoContract indigoContract
+  where
+    -- Note: eschewing optimizer here gives us more flexibility
+    compilationOptions = defaultCompilationOptions { coOptimizerConf = Nothing }
```

Comment:
Can you elaborate please? What's the flexibility gained? :thinking:

Reply:
In the optimizer, naturally. Eh, I didn't word this well, but anyway. The problem is we're running customized optimizer passes, and then `mkContract` runs default optimizer passes. This can mess things up (and does, here primarily due to https://gitlab.com/morley-framework/morley/-/issues/935, but also it's possible in general).

Reply:
I see, the new comment looks much better to me too :ok_hand: